addons/web/static/src/core/clipboard/clipboard_service.js

```diff
@@ -0,0 +1,42 @@
+/** @odoo-module **/
+
+import { registry } from "../registry";
+import { browser } from "@web/core/browser/browser";
+
+/**
+ *  @typedef {{
+ *      writeText(
+ *          String: text,
+ *          options?: ...       TODO
+ *      ): () => void;
+ *  }}
+ */
+
+export const clipboardService = {
+
+    /** @returns {void} */
+    start(env) {
+
+        function writeText(text, options = {}) {
+            setTimeout(async () => { // Use a timeout for the Safari write to clipboard to work
+                try {
+                    await browser.navigator.clipboard.writeText(text);
+                    env.services.notification.add(env._t("Link copied to clipboard."), {type: "success"});
+                    if (options.onSuccess) {
+                        options.onSuccess();
+                    }
+                } catch (error) {
+                    env.services.notification.add(env._t("Error copying to clipboard."), {type: "danger"});
```

Comment:
If this service depends on the notification service it should be one of its dependencies and receive it as such.